types/puppeteer/index.d.ts

```diff
@@ -275,13 +275,17 @@ export interface JSEvalable<A = any> {
      * The only difference between `evaluate` and `evaluateHandle` is that `evaluateHandle` returns in-page object (`JSHandle`).
      * If the function, passed to the `evaluateHandle`, returns a `Promise`, then `evaluateHandle` would wait for the
      * promise to resolve and return its value.
-     * @param fn Function to be evaluated in browser context
-     * @param args Arguments to pass to `fn`
-     */
-    evaluateHandle(
-        pageFunction: string | ((arg1: A, ...args: any[]) => any),
-        ...args: SerializableOrJSHandle[]
-    ): Promise<JSHandle>;
+     * The TypeScript definitions assume that `evaluateHandle` returns a `JSHandle`, but if you know it's going to return an
+     * `ElementHandle`, pass it as the generic argument:
+     * @param pageFunction - a function that is run within the page
+     * @param args - arguments to be passed to the pageFunction
+     */
+    // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.) # https://github.com/Microsoft/dtslint/issues/76
+    evaluateHandle<HandlerType extends JSHandle = JSHandle>(
+      pageFunction: string | ((arg1: A, ...args: any[]) => any),
+      ...args: SerializableOrJSHandle[]
+  // tslint:disable-next-line no-unnecessary-generics (This generic is meant to be passed explicitly.) # https://github.com/Microsoft/dtslint/issues/76
+  ): HandlerType extends ElementHandle ? Promise<ElementHandle> : Promise<JSHandle>;
```

Comment:
This conditional type doesn't appear to be present in the original.  Looks like it's just supposed to return `Promise<HandlerType>`?

Reply:
You're right. It looks like my mistake. I thought I had to check if the type was <code>ElementHandle</code> or <code>JSHandle</code>.  In hindsight, there was no need to do that.
Either way, the tslint exception is needed to pass the dtslint test. Should I fix and reopen PR?

Reply:
I've committed changes. Thank you.